### PR TITLE
feat: TRAC-18-Upgrade-axios-version-in-node-tracer

### PR DESCRIPTION
**Pull request by eugene-lumigo**
## Changes proposed

Upgrade to https://www.npmjs.com/package/axios/v/1.7.7

## Note to reviewers

We have a problem with axios@1.7.7 running on Node18 on Jest with jest-fetch-mock@3.0.3
We had to move part of fetch.test.js to fetchUtils.test.js to run the test without jest-fetch-mock on Node18

### CODE REVIEW

File: src/hooks/fetchUtils.test.ts [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+          },
+          expectedUrl: 'https://example.com',
+          expectedOptions: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+          },
+        },
+      ],
+      // TODO: Test FormData body
+      // TODO: Test ReadableStream body
```
Comment (harelmo-lumigo) [inline src/hooks/fetchUtils.test.ts:218]: Are these planned to be added in this PR, or just a leftover comment?

File: src/hooks/fetchUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
+import { NODE_MAJOR_VERSION } from '../../testUtils/nodeVersion';
+import { FetchInstrumentation } from './fetch';
+
+// Conditionally skip the test suite
+if (NODE_MAJOR_VERSION < 18) {
+  describe.skip('fetchUtils.skip', () => {
+    it('should not run on Node.js versions less than 18', () => {
+      // test cases here
```
Comment (harelmo-lumigo) [inline src/hooks/fetchUtils.test.ts:8]: I think we can remove this, the title says it all
```suggestion
      // test cases here
```

File: src/hooks/fetchUtils.test.ts [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    });
+  });
+} else {
```
Comment (harelmo-lumigo) [inline src/hooks/fetchUtils.test.ts:11]: I like this concept 👍

**Review by harelmo-lumigo — APPROVED** [review]

File: src/hooks/fetchUtils.test.ts [hunk 4]
```diff
@@ -0,0 +1,229 @@
+import { NODE_MAJOR_VERSION } from '../../testUtils/nodeVersion';
```
Comment (sagivoululumigo) [inline src/hooks/fetchUtils.test.ts:1]: Why did you move these tests to the `fetchUtils.test.ts` file? We usually keep the test file name the same as the file with the source code it tests (I would expect a `fetchUtils.ts` file to exist).

Please move these tests back to the `fetch.test.ts` file

Comment (sagivoululumigo) [reply src/hooks/fetchUtils.test.ts:1]: Now there is a huge diff, I can't see what is changed in the test and what isn't

File: jest.config.js [hunk 5]
```diff
@@ -17,6 +17,7 @@ module.exports = () => {
     // fetch is not supported in Node.js versions lower than 18,
     // so no need to check coverage for that version
     collectCoverageFrom.push('!./src/hooks/fetch.ts');
+    collectCoverageFrom.push('!./src/hooks/fetchUtils.ts');
```
Comment (sagivoululumigo) [inline jest.config.js:20]: There isn't such a file...

**Review by sagivoululumigo — CHANGES_REQUESTED** [review]
See comments

File: src/hooks/fetchUtils.test.ts (hunk 1 above)
Comment (eugene-lumigo) [reply src/hooks/fetchUtils.test.ts:218]: @harelmo-lumigo Leftover, I kept them while moving tests

File: jest.config.js (hunk 5 above)
Comment (eugene-lumigo) [reply jest.config.js]: Done

File: src/hooks/fetchUtils.test.ts (hunk 2 above)
Comment (eugene-lumigo) [reply src/hooks/fetchUtils.test.ts]: Done

File: src/hooks/fetchUtils.test.ts (hunk 4 above)
Comment (eugene-lumigo) [reply src/hooks/fetchUtils.test.ts:1]: Discussed that. Adding extra comments to the code.

**Review by eugene-lumigo — COMMENTED** [review]
Updated PR to reflect latest comments

**Review by harelmo-lumigo — APPROVED** [review]

File: src/hooks/fetch.test.ts [hunk 6]
```diff
@@ -3,6 +3,10 @@ import fetchMock from 'jest-fetch-mock';
 import { FetchInstrumentation } from './fetch';
 import { SpansContainer } from '../globals';
 
+// Note: With the latest version of axios, using jest-fetch-mock can cause some tests to fail.
+// To work around this issue, we've moved those specific tests to a separate file (fetchUtils.test.ts).
```
Comment (sagivoululumigo) [inline src/hooks/fetch.test.ts:7]: Specify that the parsing of headers is effected

**Review by sagivoululumigo — APPROVED** [review]
Pre-approved, just minor notes about the comments

File: src/hooks/fetchUtils.test.ts [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+import { FetchInstrumentation } from './fetch';
+
```
[the remark this answers is not in the document]
Comment (eugene-lumigo) [reply src/hooks/fetchUtils.test.ts:3]: Done

File: src/hooks/fetch.test.ts (hunk 6 above)
Comment (eugene-lumigo) [reply src/hooks/fetch.test.ts]: Done
